### DISCUSSION TITLE
Possible to use custom UnpackStrategy (in custom DownloadStrategy)?

### DISCUSSION THREAD

**Discussion by danielbayley** [Homebrew/discussions#1717] [Tap maintenance and Homebrew development]
I'm trying to create a custom `PasswordUnzipCurlDownloadStrategy` based on [`< CurlDownloadStrategy`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/download_strategy.rb#L365) (or maybe even [`NoUnzipCurlDownloadStrategy`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/download_strategy.rb#L639)) that I want to `unzip` a zip archive that has a password… How can I achieve this?

How exactly can I override the underlying `UnpackStrategy`, specifically to simply add `"-P", password` `arg`ument`s` to the array [here](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/unpack_strategy/zip.rb#L33)?

I can override the `CurlDownloadStrategy` methods inside my `PasswordUnzipCurlDownloadStrategy` `class`, but not sure if/how I can do the same with access to the `UnpackStrategy`… Also overriding the `stage` method doesn't seem to have any effect, and `unzip -o` is called (and freezes due to the zip password) regardless. Even `, using: :nounzip` doesn't prevent this!

**Comment by SMillerDev** [comment]
Tagging @Homebrew/brew for ideas

**Reply by MikeMcQuaid** [reply]
Just chiming in to say I saw this but: I have no idea really.

**Comment by danielbayley** [comment]
@SMillerDev @MikeMcQuaid
Could/should this potentially be an option added to the `container` stanza? For example something like:
~~~ rb
  container type: :zip, password: "Pa55w0rd"
~~~

**Reply by MikeMcQuaid** [reply]
I don't think so unless/until it's something we'd need in Homebrew/core or Homebrew/cask.
